### PR TITLE
[apr] Fix release only builds on Linux

### LINKED ISSUE #23488: [apr] Build error /vcpkg/packages/apr_x64-linux/debug/lib/pkgconfig/apr-1.pc   file failed to open for reading (No such file or directory):
**Host Environment**
- OS: [e.g. Windows Server 2019/ WSL Linux 20.04 LTS]
- Compiler: gcc

**To Reproduce**
```
/usr/bin/cmake -G "Ninja"   -DCMAKE_BUILD_TYPE:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="../cmake-install/WSL-GCC-Debug" -DCMAKE_TOOLCHAIN_FILE:FILEPATH="/home/user/vcpkg/scripts/buildsystems/vcpkg.cmake" -  CMakeLists.txt;
```
```
1> [CMake] *** portable models build ***
1> [CMake] -- Running vcpkg install
1> [CMake] Detecting compiler hash for triplet x64-linux...
1> [CMake] The following packages will be built and installed:
1> [CMake]   * apr[core]:x64-linux -> 1.7.0#4 -- /home/ec2-user/vcpkg/buildtrees/versioning_/versions/apr/b9e93acdaa680398eaed361f1df530096ded84ff
1> [CMake]   * apr-util[core]:x64-linux -> 1.6.1#4 -- /home/ec2-user/vcpkg/buildtrees/versioning_/versions/apr-util/cf1a0e97d00a5748dc58db7d0a5da0fd7d5efb66
1> [CMake]     log4cxx[core]:x64-linux -> 0.11.0#3 -- /home/ec2-user/vcpkg/buildtrees/versioning_/versions/log4cxx/406017355e07d94a41999c3db5e015be8bf0fbbd
1> [CMake]     tbb[core]:x64-linux -> 2020_U3#6 -- /home/ec2-user/vcpkg/buildtrees/versioning_/versions/tbb/e168e03c2e5776bdee59accf5f643ae26b0ddd84
1> [CMake]     tinyxml2[core]:x64-linux -> 9.0.0 -- /home/ec2-user/vcpkg/buildtrees/versioning_/versions/tinyxml2/a8b086bdb3eaa94fea466bf8565d7c98b00c9f94
1> [CMake] Additional packages (*) will be modified to complete this operation.
1> [CMake] Restored 0 packages from /home/ec2-user/.cache/vcpkg/archives in 1.139 ms. Use --debug to see more details.
1> [CMake] Starting package 1/5: apr:x64-linux
1> [CMake] Building package apr[core]:x64-linux...
1> [CMake] -- Installing port from location: /home/ec2-user/vcpkg/buildtrees/versioning_/versions/apr/b9e93acdaa680398eaed361f1df530096ded84ff
1> [CMake] -- Using cached apr-1.7.0.tar.bz2.
1> [CMake] -- Cleaning sources at /home/ec2-user/vcpkg/buildtrees/apr/src/apr-1-3a6fcafcf9.clean. Use --editable to skip cleaning for the packages you specify.
1> [CMake] -- Extracting source /home/ec2-user/vcpkg/downloads/apr-1.7.0.tar.bz2
1> [CMake] -- Using source at /home/ec2-user/vcpkg/buildtrees/apr/src/apr-1-3a6fcafcf9.clean
1> [CMake] -- Configuring apr
1> [CMake] -- Getting CMake variables for x64-linux-dbg
1> [CMake] -- Getting CMake variables for x64-linux-rel
1> [CMake] -- Configuring x64-linux-rel
1> [CMake] -- Building x64-linux-rel
1> [CMake] -- Installing x64-linux-rel
**1> [CMake] CMake Error at scripts/cmake/vcpkg_replace_string.cmake:12 (file):
1> [CMake]   file failed to open for reading (No such file or directory):
1> [CMake] 
1> [CMake]     /home/ec2-user/vcpkg/packages/apr_x64-linux/debug/lib/pkgconfig/apr-1.pc
1> [CMake] Call Stack (most recent call first):**
1> [CMake]   buildtrees/versioning_/versions/apr/b9e93acdaa680398eaed361f1df530096ded84ff/portfile.cmake:81 (vcpkg_replace_string)
1> [CMake]   scripts/ports.cmake:145 (include)
1> [CMake] Error: Building package apr:x64-linux failed with: BUILD_FAILED
1> [CMake] Please ensure you're using the latest portfiles with `git pull` and `./vcpkg update`.
1> [CMake] Then check for known issues at:
1> [CMake]   https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+apr
1> [CMake] You can submit a new issue at:
1> [CMake]   https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[apr]+Build+error
1> [CMake] including:
1> [CMake]   package: apr[core]:x64-linux -> 1.7.0#4
1> [CMake]     vcpkg-tool version: 2022-03-09-1affd32f93b299d5a907816c328ca3ededb73a7e
1> [CMake]     vcpkg-scripts version: 6308196b2 2022-03-09 (17 hours ago)
1> [CMake] 
1> [CMake] Additionally, attach any relevant sections from the log files above.
1> [CMake] -- Running vcpkg install - failed
1> [CMake] CMake Error at /home/ec2-user/vcpkg/scripts/buildsystems/vcpkg.cmake:834 (message):
1> [CMake]   vcpkg install failed.  See logs for more information:
1> [CMake]  /cmake-build/WSL-GCC-Debug/vcpkg-manifest-install.log
1> [CMake] Call Stack (most recent call first):
1> [CMake]   /usr/share/cmake-3.19/Modules/CMakeDetermineSystem.cmake:123 (include)
1> [CMake]   CMakeLists.txt:31 (project)
1> [CMake] CMake Error: CMake was unable to find a build program corresponding to "Ninja".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
1> [CMake] CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
1> [CMake] CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
1> [CMake] -- Configuring incomplete, errors occurred!
```
**Additional context**
Add any other context about the problem here, such as what you have already tried to resolve the issue.

### PR DESCRIPTION
**Pull request by Thomas1664**
**Describe the pull request**

- #### What does your PR fix?
  Fixes #23488

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  Unchanged

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

### CODE REVIEW

File: ports/apr/portfile.cmake [hunk 1]
```diff
@@ -84,11 +87,13 @@ else()
...
     vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/tools/${PORT}/bin/apr-1-config" "APR_BUILD_DIR=\"${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel\"" "")
-    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/tools/${PORT}/debug/bin/apr-1-config" "\"${CURRENT_INSTALLED_DIR}/debug\"" "`dirname $0`/../../../..")
-    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/tools/${PORT}/debug/bin/apr-1-config" "APR_SOURCE_DIR=\"${SOURCE_PATH}\"" "")
-    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/tools/${PORT}/debug/bin/apr-1-config" "APR_BUILD_DIR=\"${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg\"" "")
+    
+    if(NOT VCPKG_BUILD_TYPE)
+        vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/tools/${PORT}/debug/bin/apr-1-config" "\"${CURRENT_INSTALLED_DIR}/debug\"" "`dirname $0`/../../../..")
+        vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/tools/${PORT}/debug/bin/apr-1-config" "APR_SOURCE_DIR=\"${SOURCE_PATH}\"" "")
+        vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/tools/${PORT}/debug/bin/apr-1-config" "APR_BUILD_DIR=\"${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg\"" "")
+    endif()
```
Comment (Adela0814) [inline ports/apr/portfile.cmake:95]: Same as above.

File: ports/apr/portfile.cmake [hunk 2]
```diff
@@ -73,9 +73,12 @@ else()
 
     vcpkg_install_make()
 
-    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/apr-1.pc"
-        "-lapr-\${APR_MAJOR_VERSION}" "-lapr-1"
-    )
+    if(NOT VCPKG_BUILD_TYPE)
```
Comment (Adela0814) [inline ports/apr/portfile.cmake:76]: if (NOT VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")

File: ports/apr/portfile.cmake [hunk 3]
```diff
@@ -73,9 +73,12 @@ else()
...
+        vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/apr-1.pc"
+            "-lapr-\${APR_MAJOR_VERSION}" "-lapr-1"
+        )
+    endif()
+
     vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/lib/pkgconfig/apr-1.pc"
         "-lapr-\${APR_MAJOR_VERSION}" "-lapr-1"
     )
```
Comment (Adela0814) [inline ports/apr/portfile.cmake:84]: if (NOT VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "release")

**Review by Adela0814 — CHANGES_REQUESTED** [review]

**Review by dg0yt — COMMENTED** [review]
In portfiles, `VCPKG_BUILD_TYPE` can be false or `release`. Single-config `debug` is not supported. So the `if` code is okay. It is also accepted in all recent PRs.

**Comment by Adela0814** [comment]
> In portfiles, `VCPKG_BUILD_TYPE` can be false or `release`. Single-config `debug` is not supported. So the `if` code is okay. It is also accepted in all recent PRs.

ok, got it.

**Review by Adela0814 — APPROVED** [review]

**Comment by strega-nil-ms** [comment]
Thanks!

**Pull request merged**